conan/cli/commands/profile.py

```diff
@@ -9,35 +9,44 @@
 from conans.util.files import save
 
 
-def print_profiles(profiles):
-    host, build = profiles
-    cli_out_write("Host profile:")
-    cli_out_write(host.dumps())
-    cli_out_write("Build profile:")
-    cli_out_write(build.dumps())
+def _print_profiles(profiles):
+    if "host" in profiles:
+        ConanOutput().info("Host profile:")
+        cli_out_write(profiles["host"].dumps())
+    if "build" in profiles:
+        ConanOutput().info("Build profile:")
+        cli_out_write(profiles["build"].dumps())
 
 
 def profiles_list_cli_output(profiles):
-    cli_out_write("Profiles found in the cache:")
+    ConanOutput().info("Profiles found in the cache:")
     for p in profiles:
         cli_out_write(p)
 
 
-def json_profiles(profiles):
-    host, build = profiles
-    result = {"host": host.serialize(),
-              "build": build.serialize()}
+def _json_profiles(profiles):
+    result = {}
+    if "host" in profiles:
+        result["host"] = profiles["host"].serialize()
+    if "build" in profiles:
+        result["build"] = profiles["build"].serialize()
     cli_out_write(json.dumps(result))
 
 
-@conan_subcommand(formatters={"text": print_profiles, "json": json_profiles})
+@conan_subcommand(formatters={"text": _print_profiles, "json": _json_profiles})
 def profile_show(conan_api, parser, subparser, *args):
     """
     Show aggregated profiles from the passed arguments.
     """
     add_profiles_args(subparser)
+    subparser.add_argument("--context", choices=["host", "build"])
```

Comment:
Wdyt?

```suggestion
    subparser.add_argument("-cx", "--context", choices=["host", "build"])
```

Reply:
Good idea - will let the tests pass in this run before merging this sugegstion :)